File: server/Microservices/MovieService/MovieService.Domain/Models/SeatModel.cs
namespace MovieService.Domain.Models;

public class SeatModel
{
	public Guid Id { get; set; }
	public Guid HallId { get; set; }
	public Guid SeatTypeId { get; set; }
	public int Row { get; set; }
	public int Column { get; set; }

	public SeatModel()
	{
	}

	public SeatModel(
		Guid id,
		Guid hallId,
		Guid seatTypeId,
		int row,
		int column)
	{
		Id = id;
		HallId = hallId;
		SeatTypeId = seatTypeId;
		Row = row;
		Column = column;
	}
}

File: src/mobile/lib/data/models/session/seat_model.dart
// lib/data/models/session/seat_model.dart
import '../../../domain/entities/session/seat.dart';

class SeatModel extends Seat {
  const SeatModel({required String id, required int row, required int column})
    : super(id: id, row: row, column: column);

  factory SeatModel.fromJson(Map<String, dynamic> json) {
    return SeatModel(id: json['id'], row: json['row'], column: json['column']);
  }

  Map<String, dynamic> toJson() {
    return {'id': id, 'row': row, 'column': column};
  }
}
